benches/tree_hash_root.rs

```diff
@@ -43,6 +43,26 @@ pub fn tree_hash_root(c: &mut Criterion) {
             })
         },
     );
+
+    c.bench_with_input(
+        BenchmarkId::new("tree_hash_root_list_parallel", size),
+        &size,
+        |b, &size| {
+            b.iter(|| {
+                let l1 = List::<u64, C>::try_from_iter(0..size).unwrap();
+                let mut l2 = l1.clone();
+                l2.push(99).unwrap();
+                l2.apply_updates().unwrap();
+
+                let handle = std::thread::spawn(move || {
+                    l1.tree_hash_root();
+                    l2.tree_hash_root();
+                });
```

Comment:
To run in paralell I think we should spawn _two_threads with two calls to `std::thread::spawn`, and hash `l1` in one and `l2` in the other